File: Backend/Models/Entities/Sole_Supplier.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProcionAPI.Models.Entities
{
    public class Sole_Supplier
    {
        [Key]
        public int Sole_Supplier_ID { get; set; }
        [Required]
        public int Vendor_ID { get; set; }
        [ForeignKey("Vendor_ID")]
        public Vendor Vendor { get; set; }
        public bool MD_Approval { get; set; }
        public DateTime Date { get; set; }
        public string Reason { get; set; }
    }
}

File: IonicFrontend/Procionic/src/app/Shared/Sole_Supplier.ts
import { VendorOnboardRequest } from "./VendorOnboardRequest";


export interface SoleSupplier {

    sole_Supplier_ID: number;
    vendor_ID: number;
    vendor: VendorOnboardRequest
    mD_Approval : boolean;
    date: Date;
    reason : string;
}